tools/DocsLinter/SimplifiedMarkdownDoc.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+
+namespace DocsLinter
+{
+  /// <summary>
+  ///   Simple data class that contains information related to a Markdown doc that we require for linting.
+  /// </summary>
+  public class SimplifiedMarkdownDoc
+  {
+    /// <summary>
+    ///   List of links in the markdown file
+    /// </summary>
+    public List<LinkBase> Links = new List<LinkBase>();
+
+    /// <summary>
+    ///   List of headings in the markdown file
+    /// </summary>
+    public List<Heading> Headings = new List<Heading>();
+
+    public SimplifiedMarkdownDoc()
+    {
+    }
+
+    /// <summary>
+    ///   Constructor for automatically parsing a MarkdownDocument object from Markdig.
+    /// </summary>
+    /// <param name="markdownDoc"></param>
+    public SimplifiedMarkdownDoc(MarkdownDocument markdownDoc)
+    {
+      Links.AddRange(markdownDoc.Descendants().OfType<LinkInline>().Select(ParseLink));
+      Headings.AddRange(markdownDoc.Descendants().OfType<HeadingBlock>().Select(heading => new Heading(heading)));
+    }
+
+    /// <summary>
+    ///   A helper function that parses the Markdig link object and returns the correct type of link - remote or local.
+    /// </summary>
+    /// <param name="linkInline">The object from Markdig to parse.</param>
+    /// <returns></returns>
+    public static LinkBase ParseLink(LinkInline linkInline)
+    {
+      if (linkInline.Url.StartsWith("http") || linkInline.Url.StartsWith("www"))
+      {
+        return new RemoteLink(linkInline);
+      }
+
+      return new LocalLink(linkInline);
+    }
+  }
+
+  /// <summary>
+  ///   Abstract base class for a link.
+  /// </summary>
+  public abstract class LinkBase
+  {
+  }
+
+  /// <summary>
+  ///   A data class that represents a remote link. I.e. - "https://www.google.com".
+  /// </summary>
+  public class RemoteLink : LinkBase
+  {
+    /// <summary>
+    ///   The URL of the link.
+    /// </summary>
+    public string Url;
+
+    /// <summary>
+    ///   Constructor for that parses the Markdig link object.
+    /// </summary>
+    /// <param name="link">The Markdig link object.</param>
+    public RemoteLink(LinkInline link)
+    {
+      Url = link.Url;
+    }
+
+    public override string ToString()
+    {
+      return Url;
+    }
+  }
+
+  /// <summary>
+  ///   A data class that represents a local link. I.e. - "../README.md#heading"
+  /// </summary>
+  public class LocalLink : LinkBase
```

Comment:
If LinkBase is a interface then can this be a struct as this just looks like it's data.